src/main/java/org/openrewrite/analysis/trait/variable/LocalVariableDecl.java

```diff
@@ -105,7 +105,7 @@ static Validation<TraitErrors, Callable> findNearestParentCallable(Cursor cursor
         while (path.hasNext()) {
             Cursor c = path.next();
             Tree t = c.getValue();
-            if (t instanceof J.ClassDeclaration || t instanceof J.NewClass) {
+            if (t instanceof J.ClassDeclaration || t instanceof J.NewClass && previous != null && ((J.NewClass) t).getBody() == previous.getValue()) {
                 break;
             }
             if (t instanceof J.Block && J.Block.isStaticOrInitBlock(c)) {
```

Comment:
Also note that in the context of lambdas the body block was erroneously identified as an initializer block. That is fixed by https://github.com/openrewrite/rewrite/commit/69b48c8338da56969fb0a9433b6dea47adb28737

Reply:
Much appreciated! Thanks for fixing!